scipy/odr/tests/test_odr.py

```diff
@@ -524,3 +524,9 @@ def func(b, x):
             # remove output files for clean up
             shutil.rmtree(tmp_dir)
 
+    def test_odr_model_default_meta(self):
+        def func(b, x):
+            return b[0] + b[1] * x
+
+        p = Model(func)
+        assert_equal(p.meta, {})
```

Comment:
Instead I would test going through the iterator as in the issue.

Reply:
@tupui  I'm not sure how to do the test using iterator, can you show me an example please?

Reply:
The `.meta` attribute being a `dict` is part of the public interface, not just an internal implementation detail. Testing that it is equal to `{}` in this case is a perfectly good thing to test. But it also would be a good idea to test that the other public functionality on `Model` that internally accesses `.meta` works, too. So test that `__getattr__()` behaves appropriately and `set_meta()`.

Reply:
@rkern  got it, thanks , updated the test  as well as the code.